src/main/java/net/dv8tion/jda/core/handle/EventCache.java

```diff
@@ -77,6 +77,19 @@ public int size()
                 .sum();
     }
 
+    public void clear(long triggerId, Type type)
+    {
+        TLongObjectMap<List<Runnable>> cache = eventCache.get(type);
+        if (cache == null)
+            return;
+        List<Runnable> events = cache.get(triggerId);
+        if (events == null)
+            return;
+        LOG.debug("Clearing cache for type " + type + " with ID " + triggerId + " (Size: " + events.size() + ')');
+        events.clear();
+        cache.remove(triggerId);
```

Comment:
change `cache.get` to `cache.remove`. Remove has the item removed returned. This will also prevent possible raceconditions.